src/pybreaker.py

```diff
@@ -214,6 +230,98 @@ def remove_listener(self, listener):
             self._listeners.remove(listener)
 
 
+class CircuitBreakerStorage(object):
+    def __init__(self, name):
+        """
+        """
+        self._name = name
+
+    @property
+    def name(self):
+        return name
+
+    @property
+    def state(self):
+        pass
+
+    @state.setter
+    def state(self, state):
+        pass
+
+    def increment_counter(self):
+        pass
+
+    def reset_counter(self):
+        pass
+
+    @property
+    def counter(self):
+        pass
+
+    @property
+    def opened_at(self):
+        pass
+
+    @state.setter
+    def opened_at(self, datetime):
+        pass
+
+
+class CircuitMemoryStorage(CircuitBreakerStorage):
+    def __init__(self, state):
+        """
+        """
+        super(CircuitMemoryStorage, self).__init__('memory')
+        self._fail_counter = 0
+        self._opened_at = None
+        self._state = state
+
+    @property
+    def state(self):
+        return self._state
+
+    @state.setter
+    def state(self, state):
+        self._state = state
+
+    def increment_counter(self):
+        self._fail_counter += 1
+
+    def reset_counter(self):
+        self._fail_counter = 0
+
+    @property
+    def counter(self):
+        return self._fail_counter
+
+    @property
+    def opened_at(self):
+        return self._opened_at
+
+    @state.setter
+    def opened_at(self, datetime):
+        self._opened_at = datetime
+
+
+class CircuitRedisStorage(object):
```

Comment:
@phillbaker remember to only allow users to use this storage backend if its dependencies are also installed.